pep-0517.txt

```diff
@@ -244,28 +185,33 @@ metadata. The directory passed in by the build frontend MUST be
 identical to the directory created by ``prepare_wheel_metadata``,
 including any unrecognized files it created.
 
-This must return the basename (not the full path) of the ``.whl`` file it
+If build_directory is not None, it is a unicode string containing the
+path to a directory where intermediate build artifacts may be stored.
+This may be empty, or it may contain artifacts from a previous build to
+be used as a cache. The backend is responsible for determining whether
+any cached artifacts are outdated. When a build_directory is provided,
+the backend should not create or modify any files in the source
+directory (the working directory where the hook is called). If the
+backend cannot reliably avoid modifying the directory it builds from, it
+should copy any files it needs to build_directory and perform the build
+there.
+
+If build_directory is None, the backend may do an 'in place' build which
+modifies the source directory. The semantics of this are not specified
+here.
+
+Whatever the value of build_directory, the backend may also store intermediates
+in other cache locations or temporary directories, which it is responsible for
+managing. The presence or absence of any caches should not make a
+material difference to the final result of the build.
+
+The hook must return the basename (not the full path) of the ``.whl`` file it
 creates, as a unicode string.
```

Comment:
Given the length of this hook description now, perhaps the description of the return value should be moved ahead of the definition of the input parameters? Alternatively, perhaps just repeat where the wheel gets created: "...it creates in ``wheel_directory``, as a..."